ipaserver/install/ipa_otptoken_import.py

```diff
@@ -506,6 +508,95 @@ def save(self, dest):
         self.__doc.write(dest)
 
 
+class CSVToken(object):
+    @property
+    def id(self):
+        return self.__id
+
+    @property
+    def options(self):
+        return self.__options
+
+    @property
+    def removed(self):
+        return self.__removed
+
+    def __init__(self, id, options):
+        self.__id = id
+        self.__options = options
+        self.__removed = False
+
+    def remove(self):
+        self.__removed = True
+
+
+class CSVDocument(object):
+    _FIELDS = ("serial", "key", "type", "digits", "interval")
```

Comment:
Why did you define different field names from IPA field names? This trick should make the code faster, smaller, and easier to understand:

```
_MAP = collections.OrderedDict([
    ('ipatokenserial', lambda v, o: v.strip()),
    ...
])

with open(...) as f:
    reader = csv.DictReader(f, list(self._MAP), restval='')
    for row in reader:
        for k, v in row.items():
            row[k] = self._MAP[k](v, options)
```